### PR TITLE
feat: Add infos about dnf-plugins-core

### PR DESCRIPTION
**Pull request by alemorvan**
#### Author checklist (Completed by original Author)
- [x] Good fit for the Rocky Linux project? Title and Author Metatags inserted ?
- [x] If applicable, steps and instructions have been tested to work
- [x] Initial self-review to fix basic typos and grammar completed

#### Rocky Documentation checklist (Completed by Rocky team) 
- [x] 1st Pass (Document is good fit for project and author checklist completed)
- [x] 2nd Pass (Technical Review - check for technical correctness) 
- [x] 3rd Pass (Detailed Editorial Review and Peer Review)
- [x] Final approval (Final Review)

### CODE REVIEW

File: docs/books/admin_guide/13-softwares.md [hunk 1]
```diff
@@ -900,3 +900,121 @@ Is this ok [y/N]:
...
+
+Or if you want to also download the dependencies:
+
+```
+dnf download --resolv --alldeps ansible
+```
+
+### `needs-restarting` plugin
+
+After running an `dnf update`, the running processes will continue to run but with the old binaries. In order to take into account the code changes and especially the security updates, they have to be restarted.
```
Comment (SergeCroise) [inline docs/books/admin_guide/13-softwares.md:982]:
```suggestion
After running a `dnf update`, the running processes will continue to run but with the old binaries. In order to take into account the code changes and especially the security updates, they have to be restarted.
```

Comment (NezSez) [reply docs/books/admin_guide/13-softwares.md:982]: I'd add for after a `dnf update` (or any package update really) :
`rpmconf -a -c` (not sure if this pkg is default or from EPEL)
`dnf needs-restarting -s` to check if services need restarting
`dnf needs-restarting -r` to check if a reboot may be required
`dnf needs-restarting -s -r` to do both in one run

**Comment by alemorvan** [comment]
Thx @SergeCroise I commited your change

File: docs/books/admin_guide/13-softwares.md (hunk 1 above)
Comment (alemorvan) [reply docs/books/admin_guide/13-softwares.md]: @NezSez:
I want to keep as simple as possible this admin guide. I think your use case for rpmconf -a -c (never needed to use it personnaly) is maybe something that should not happened? In that case, a guide would be better.

Comment (alemorvan) [reply docs/books/admin_guide/13-softwares.md]: @NezSez your descriptions for needs-restarting are better than mine.
Please start a review in the github UI and make a suggestion as Serge did just before you.
Like that, the changes will be commited and will belong to you in a git-blame

Comment (NezSez) [reply docs/books/admin_guide/13-softwares.md]: Yes, I'm familiar with git, not so much GitHub, and I didn't have time to figure out it's interface, so I just made the comment.  I also haven't read this admin book so not familiar with context and figured you or Serge would know what to do with the info.

File: docs/books/admin_guide/13-softwares.md [hunk 2]
```diff
@@ -900,3 +900,121 @@ Is this ok [y/N]:
...
 
 EPEL is not an official repository for RHEL. But it can be useful for administrators and developers who work with RHEL or derivatives and need some utilities prepared for RHEL from a source they can feel confident about.
+
+## DNF Plugins
+
+The `dnf-plugins-core` package adds plugins to `dnf` that will be useful for managing your repositories.
+
+!!! NOTE
+    See more informations here: https://dnf-plugins-core.readthedocs.io/en/latest/index.html
+
```
Comment (sspencerwire) [inline docs/books/admin_guide/13-softwares.md:910]: Hey @alemorvan, for the most part this looks good. I didn't have a chance to test anything and no time to really look hard at it. I *think* that you and I decided that for admonitions we needed the content to be one line down from the admonition type like this:
```
!!! Note
    
    See more information here ...
```
If I recall, this had something to do with how Crowdin messed with admonitions.

File: docs/books/admin_guide/13-softwares.md [hunk 3]
```diff
@@ -900,3 +900,121 @@ Is this ok [y/N]:
...
+### `versionlock` plugin
+
+Sometimes it is useful to protect packages from all updates or to exclude certain versions of a package (because of known problems for example). For this purpose, the versionlock plugin will be of great help. 
+
+You need to install an extra package:
+
+```
+python3-dnf-plugin-versionlock.noarch
+```
+
```
Comment (sspencerwire) [inline docs/books/admin_guide/13-softwares.md:1005]: Line 1003... is that correct? (again, no time to test it out at the moment) or should it be `dnf install python3-dnf-plugin-versionlock`?

File: docs/books/admin_guide/13-softwares.md [hunk 4]
```diff
@@ -900,3 +900,121 @@ Is this ok [y/N]:
...
+dnf versionlock add ansible
+Adding versionlock on: ansible-0:6.3.0-2.el9.*
+```
+
+* List locked packages:
+
+```
+dnf versionlock list
+ansible-0:6.3.0-2.el9.*
+```
```
Comment (sspencerwire) [inline docs/books/admin_guide/13-softwares.md:1020]: Overall, I don't see problems with language issues. Perhaps @SergeCroise or @Grammaresque can look it over from a different perspective. Thanks for your work!!

File: docs/books/admin_guide/13-softwares.md [hunk 5]
```diff
@@ -900,3 +900,121 @@ Is this ok [y/N]:
...
+The `needs-restarting` plugin will allow you to detect processes that are in this case.
+
+```
+dnf needs-restarting [-u] [-r] [-s]
+```
+
+| Options | Description                                                                  |
+| --------| --------------------------------------------------------------------------- |
+| `-u`     | Only consider processes belonging to the running user.                       |
+| `-r`     | Only report whether a reboot is required (exit code 1) or not (exit code 0). |
```
Comment (alemorvan) [inline docs/books/admin_guide/13-softwares.md:993]:
```suggestion
| `-r`     |  to check if a reboot may be required |
```

File: docs/books/admin_guide/13-softwares.md [hunk 6]
```diff
@@ -900,3 +900,121 @@ Is this ok [y/N]:
...
+| `-s`     | Only list the affected systemd services.                                     |
```
Comment (alemorvan) [inline docs/books/admin_guide/13-softwares.md:994]:
```suggestion
| `-s`     | to check if services need restarting                                  |
| `-s -r` | to do both in one run                                                       |
```

File: docs/books/admin_guide/13-softwares.md [hunk 7]
```diff
@@ -900,3 +900,121 @@ Is this ok [y/N]:
 ### Conclusion
 
 EPEL is not an official repository for RHEL. But it can be useful for administrators and developers who work with RHEL or derivatives and need some utilities prepared for RHEL from a source they can feel confident about.
+
+## DNF Plugins
+
+The `dnf-plugins-core` package adds plugins to `dnf` that will be useful for managing your repositories.
+
+!!! NOTE
```
Comment (alemorvan) [inline docs/books/admin_guide/13-softwares.md:908]:
```suggestion
!!! NOTE

```

File: docs/books/admin_guide/13-softwares.md (hunk 2 above)
Comment (alemorvan) [reply docs/books/admin_guide/13-softwares.md:911]: I did the change.
Thx

File: docs/books/admin_guide/13-softwares.md [hunk 8]
```diff
@@ -900,3 +900,123 @@ Is this ok [y/N]:
...
+| `-s -r` | to do both in one run.                                                       |
+
+### `versionlock` plugin
+
+Sometimes it is useful to protect packages from all updates or to exclude certain versions of a package (because of known problems for example). For this purpose, the versionlock plugin will be of great help. 
+
+You need to install an extra package:
+
+```
+python3-dnf-plugin-versionlock.noarch
```
Comment (alemorvan) [inline docs/books/admin_guide/13-softwares.md:1005]:
```suggestion
dnf install python3-dnf-plugin-versionlock
```

**Review by sspencerwire — APPROVED** [review]
This looks good! Thanks @alemorvan

**Comment by sspencerwire** [comment]
I appreciate you taking a look at this @SergeCroise !

**Pull request merged**